installer/arch/prepare

```diff
@@ -0,0 +1,244 @@
+#!/bin/sh -e
+# Copyright (c) 2014 The crouton Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This is a distro-specific continuation of the prepare.sh script.
+
+PKGEXT='tar.xz'
+DISTROAKA='arch'
+
+# aurcompile <package> <extra makepkg options>: compile package from AUR.
+# This should not be a permanent solution, as AUR packages cannot be trusted,
+# and may break. Ideally, we would like packages to be pushed to [community].
+# We create the user 'aur' to compile the package, giving it sudo access, so
+# that dependencies and the package itself can be installed automatically.
+aurcompile() {
+    local package="$1"
+    shift
+    echo "Compiling $package from AUR..." 1>&2
+    # Make sure base-devel and wget are installed
+    # Expand base-devel so that install can replace packages with their
+    # multilib versions
+    install --asdeps `pacman -Sg base-devel | cut -d' ' -f 2`
+    install wget ca-certificates
+
+    if ! id aur > /dev/null 2>&1; then
+        useradd aur -r -s /bin/false
+    fi
+
+    SUDOERSFILE="/etc/sudoers.d/99-aur-pacman"
+    echo "aur ALL = NOPASSWD:/usr/bin/pacman" > "$SUDOERSFILE"
+    addtrap "rm -f '$SUDOERSFILE'"
+
+    su aur -s /bin/sh -c "
+        set -e
+        PACKAGE='$package'"'
+        PACKAGE2="${PACKAGE%"${PACKAGE#??}"}"
+        TMPDIR="`mktemp -d --tmpdir=/tmp crouton-aur.XXX`"
+        TRAP="rm -rf '"'"'$TMPDIR'"'"'"
+        trap "$TRAP" INT HUP TERM 0
+        cd "$TMPDIR"
+        wget "https://aur.archlinux.org/packages/$PACKAGE2/$PACKAGE/$PACKAGE.tar.gz"
+        tar vxf "$PACKAGE.tar.gz"
+        cd "$PACKAGE"
+        pkgver=`sed -n '"'"'s/^pkgver=//p'"'"' PKGBUILD`
+        pkgrel=`sed -n '"'"'s/^pkgrel=//p'"'"' PKGBUILD`
+        # FIXME: Temporary fix for mawk (the AUR PKGBUILD currently works, but
+        # will break again on the next upstream update)
+        if [ "$PACKAGE" = "mawk" ]; then
```

Comment:
The "bug report" is the my comment under `nboichat` there: https://aur.archlinux.org/packages/mawk/ .

Anyway, we do not need `mawk` anymore, dropping the hack.